chainindex/indexer.go

```diff
@@ -338,12 +374,18 @@ func (si *SqliteIndexer) Revert(ctx context.Context, from, to *types.TipSet) err
 	}
 
 	err = withTx(ctx, si.db, func(tx *sql.Tx) error {
+		// revert the `from` tipset
 		if _, err := tx.Stmt(si.updateTipsetToRevertedStmt).ExecContext(ctx, revertTsKeyCid); err != nil {
 			return xerrors.Errorf("error marking tipset %s as reverted: %w", revertTsKeyCid, err)
 		}
 
+		// index the `to` tipset -> it is idempotent
```

Comment:
@rvagg I really think this is great for auto-repair etc and is idempotent and no harm in keeping it here.